### ISSUE TITLE
Feature Request: Custom Autocmd Event on Reload Theme

### ISSUE THREAD

**Issue by shaun-mathew** [NvChad/NvChad#1459]
**Is your feature request related to a problem? Please describe.**
I'd like to run some custom logic whenever NvChad reloads or changes its theme.

**Describe the solution you'd like**
A simple user autocmd event would like NvChadThemeLoaded: 
```vim
doautocmd User NvChadThemeLoaded
```

**Describe alternatives you've considered**
I've consider using the vanilla ColorScheme autocmd, but NvChad doesn't trigger this event when changing themes or cycling
through themes in NvChad's telescope themes.

**Comment by siduck** [comment]
idk much about this , so this has to run every time we reload the theme?
```
doautocmd User NvChadThemeLoaded
```

**Comment by shaun-mathew** [comment]
Yes, although I am not sure of the neovim lua api call for this.

See: https://stackoverflow.com/questions/21535350/creating-a-user-defined-event-in-vim

**Comment by siduck** [comment]
im confused! do we have to define the autocmd whenever reloading the theme?
